music21/spanner.py

```diff
@@ -1253,16 +1318,22 @@ def setPendingSpannedElementAssignment(
         self,
         sp: Spanner,
         className: str,
+        offsetInScore: OffsetQL|None = None,
+        clientInfo: t.Any|None = None
     ):
         '''
-        A SpannerBundle can be set up so that a particular spanner (sp)
-        is looking for an element of class (className) to complete it. Any future
-        element that matches the className which is passed to the SpannerBundle
-        via freePendingSpannedElementAssignment() will get it.
+        A SpannerBundle can be set up so that a particular spanner (sp) is looking
+        for an element of class (className) to be set as first element. Any future
+        future element that matches the className (and offsetInScore, if specified)
+        which is passed to the SpannerBundle via freePendingFirstSpannedElementAssignment()
+        will get it.  clientInfo is not used in the match, but can be used by the client
+        when cleaning up any leftover pending assignments, by creating SpannerAnchors
+        at the appropriate offset.
```

Comment:
please demo why we would want this.

Reply:
Will do.

Reply:
Done.  With much more commentary as well.